source/s3_request_messages.c

```diff
@@ -169,11 +172,12 @@ struct aws_http_message *aws_s3_create_multipart_upload_message_new(
     AWS_PRECONDITION(allocator);
 
     /* For multipart upload, sse related headers should only be shown in create-multipart request */
-    struct aws_http_message *message = aws_s3_message_util_copy_http_message_no_body(
+    struct aws_http_message *message = aws_s3_message_util_copy_http_message_no_body_filter_headers(
```

Comment:
trivial: can we add a comment here, something like "Only include the x-amz-meta-* when the object/upload created"? It was clear before, as we have different list of headers to exclude for different type of request, but when we include `x-amz-meta-*` is quite unclear